src/app-layout/__integ__/awsui-applayout.test.ts

```diff
@@ -167,11 +167,19 @@ describe.each(['classic', 'refresh', 'refresh-toolbar'] as Theme[])('%s', theme
   test(
     'does not render notifications slot when it is empty',
     setupTest({ pageName: 'with-notifications' }, async page => {
-      const { height: originalHeight } = await page.getBoundingBox(wrapper.findNotifications().toSelector());
-      expect(originalHeight).toBeGreaterThan(0);
+      const { height: headerHeight } = await page.getBoundingBox('#h');
       await page.click(wrapper.findNotifications().findFlashbar().findItems().get(1).findDismissButton().toSelector());
-      const { height: newHeight } = await page.getBoundingBox(wrapper.findNotifications().toSelector());
-      expect(newHeight).toEqual(0);
+      await expect(page.isExisting(wrapper.findNotifications().findFlashbar().toSelector())).resolves.toBe(true);
+      await expect(
+        page.getElementsCount(wrapper.findNotifications().findFlashbar().findItems().toSelector())
+      ).resolves.toBe(0);
+      const { top: contentTop } = await page.getBoundingBox('[data-testid="content-root"]');
+      const expectedOffset = {
+        classic: headerHeight + 16,
+        refresh: headerHeight + 12,
+        'refresh-toolbar': headerHeight + 54,
+      }[theme];
+      expect(contentTop).toEqual(expectedOffset);
```

Comment:
Isn't it be easier to only check notification container height and make sure it equals 0?

Reply:
This is how original test was structured and it still allowed this regression to happen.

The new approach makes sure this test fails every time there is an extended top offset of the main content, regardless how it is implemented internally